core/lib/vlog/src/lib.rs

```diff
@@ -23,7 +25,36 @@ pub struct ObservabilityBuilder {
 /// Guard for the observability subsystem.
 /// Releases configured integrations upon being dropped.
 pub struct ObservabilityGuard {
-    _sentry_guard: Option<ClientInitGuard>,
+    /// Opentelemetry provider. Can be used to force flush spans.
+    otlp_provider: Option<opentelemetry_sdk::trace::TracerProvider>,
+    sentry_guard: Option<ClientInitGuard>,
+}
+
+impl ObservabilityGuard {
+    /// Forces flushing of pending events.
+    /// This method is blocking.
+    pub fn force_flush(&self) {
+        // We don't want to wait for too long.
+        const FLUSH_TIMEOUT: Duration = Duration::from_secs(1);
+
+        if let Some(sentry_guard) = &self.sentry_guard {
+            sentry_guard.flush(Some(FLUSH_TIMEOUT));
+        }
+
+        if let Some(provider) = &self.otlp_provider {
+            for result in provider.force_flush() {
```

Comment:
Hmm, looks like otlp examples [suggest](https://github.com/open-telemetry/opentelemetry-rust/blob/main@%7B2024-05-20T12:41:02Z%7D/opentelemetry-otlp/examples/basic-otlp/src/main.rs#L137-L139) using `shutdown`. Is there a reason to call `force_flush` instead?

Reply:
Ah, looks like they are doing different things... Maybe let's call both then?